src/blockchain/Blockchain.ts

```diff
@@ -47,56 +52,105 @@ export interface IBlockchainApi {
+  )
+}
 export const EXTRINSIC_FAILED: ResultEvaluator = (result) =>
   ErrorHandler.extrinsicFailed(result) &&
   (ErrorHandler.getExtrinsicError(result) || UNKNOWN_EXTRINSIC_ERROR)
 
 /**
- * Submits a signed [[SubmittableExtrinsic]] and attaches a callback to monitor the inclusion status of the transaction
+ * Parses potentially incomplete or undefined options and returns complete [[SubscriptionPromiseOptions]].
+ *
+ * @param opts Potentially undefined or partial [[SubscriptionPromiseOptions]] .
+ * @returns Complete [[SubscriptionPromiseOptions]], with potentially defaulted values.
+ */
+export function parseSubscriptionOptions(
+  opts?: Partial<SubscriptionPromiseOptions>
+): SubscriptionPromiseOptions {
+  const {
+    resolveOn = IS_FINALIZED,
+    rejectOn = (result: SubmittableResult) =>
+      IS_ERROR(result) || EXTRINSIC_FAILED(result) || IS_USURPED(result),
+    timeout,
+  } = { ...opts }
+  return { resolveOn, rejectOn, timeout }
+}
+
+/**
+ * [ASYNC] Submits a signed [[SubmittableExtrinsic]] and attaches a callback to monitor the inclusion status of the transaction
  * and possible errors in the execution of extrinsics. Returns a promise to that end which by default resolves upon
  * finalization and rejects any errors occur during submission or execution of extrinsics. This behavior can be adjusted via optional parameters.
  *
  * Transaction fees will apply whenever a transaction fee makes it into a block, even if extrinsics fail to execute correctly!
  *
  * @param tx The [[SubmittableExtrinsic]] to be submitted. Most transactions need to be signed, this must be done beforehand.
- * @param resolveOn A function which triggers the resolution of the promise. Defaults to resolution on finalization.
- * @param rejectOn A function which triggers the rejection of the promise and specifies the rejection reason.
- * Defaults to rejection if either the submission of the transaction failed or if the execution of extrinsics emitted an error event.
- * @param timeout Optional timeout in ms. If set, an unresolved promise will reject after this period of time.
+ * @param opts [[SubscriptionPromiseOptions]]: Criteria for resolving/rejecting the promise.
  * @returns A promise which can be used to track transaction status.
  * If resolved, this promise returns [[SubmittableResult]] that has led to its resolution.
  */
 export async function submitSignedTx(
   tx: SubmittableExtrinsic,
-  resolveOn: ResultEvaluator = IS_FINALIZED,
-  rejectOn: ResultEvaluator = (result) =>
-    IS_ERROR(result) || EXTRINSIC_FAILED(result),
-  timeout?: number
+  opts: SubscriptionPromiseOptions
 ): Promise<SubmittableResult> {
-  log.info(`Submitting ${tx.method}`)
-  const { promise, subscription } = makeSubscriptionPromise(
-    resolveOn,
-    rejectOn,
-    timeout
-  )
-  const unsubscribe = await tx.send(subscription)
-  return promise.finally(() => unsubscribe())
+  const { promise, subscription } = makeSubscriptionPromise(opts)
```

Comment:
let's keep the log

Reply:
Was placed in wrong place, now back to original